src/gridtypes/unstructured.jl

```diff
@@ -75,22 +80,31 @@ function vtk_grid{T<:AbstractFloat}(
     return vtk::DatasetFile
 end
 
-
-# Variant of vtk_grid with 1-D arrays x, y, z.
-# Size of each array: (num_points)
 function vtk_grid{T<:AbstractFloat}(
         filename_noext::AbstractString,
-        x::AbstractArray{T}, y::AbstractArray{T}, z::AbstractArray{T},
-        cells::Vector{MeshCell};
+        points::AbstractArray{T,N}, cells::Vector{MeshCell};
         compress::Bool=true, append::Bool=true)
-    @assert length(x) == length(y) == length(z)
-    Npts = length(x)
-    points = Array(T, 3, Npts)
-    for n = 1:Npts
-        points[1, n] = x[n]
-        points[2, n] = y[n]
-        points[3, n] = z[n]
+
+    # Check dimension
+    if N == 2
+        dim, Ni = size(points)
+        Nj, Nk = 1,1
+    elseif N == 3
+        dim, Ni, Nj = size(points)
+        Nk = 1
+    elseif N == 4
+        dim, Ni, Nj, Nk = size(points)
+    else
+        throw(ArgumentError("points must be of size (dim, Npts_x; Npts_y, Npts_z)"))
     end
-    return vtk_grid(filename_noext, points, cells;
-                    compress=compress, append=append)::DatasetFile
+
+    return vtk_grid(filename_noext, reshape(points,(dim,Ni*Nj*Nk)), cells, compress=compress, append=append)
 end
+
+@deprecate(vtk_grid{T<:AbstractFloat}(
```

Comment:
Deprecated this since I didn't wanna write new functions for 1 and 2D, so much copy paste, and its just as easy to gather all the points in one array.